Hymns/Display/VerseView.swift

```diff
@@ -0,0 +1,65 @@
+import SwiftUI
+
+struct VerseLineView: View {
+    private let verseNumber: String?
+    private let verseText: String
+
+    init(verseNumber: String, verseText: String) {
+        self.verseNumber = verseNumber
+        self.verseText = verseText
+    }
+
+    init(verseText: String) {
+        self.verseNumber = nil
+        self.verseText = verseText
+    }
+
+    var body: some View {
+        HStack {
+            Text(verseNumber ?? "").frame(width: 10)
+            Text(verseText)
+        }
+    }
+}
+
+struct VerseLineView_Previews: PreviewProvider {
+    static var previews: some View {
+        VerseLineView(verseNumber: "1", verseText: "Drink! A river pure and clear that's flowing from the throne")
+    }
+}
+
+struct VerseView: View {
+    private let verseNumber: String?
+    private let verseLines: [String]
+
+    init(verseNumber: String, verseLines: [String]) {
+        self.verseNumber = verseNumber
+        self.verseLines = verseLines
+    }
+
+    init(verseLines: [String]) {
+        self.verseNumber = nil
+        self.verseLines = verseLines
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            ForEach(0..<verseLines.count) { index in
+                if self.verseNumber != nil && index == 0 {
+                    VerseLineView(verseNumber: self.verseNumber!, verseText: self.verseLines[index])
+                } else {
+                    VerseLineView(verseText: self.verseLines[index])
+                }
```

Comment:
Admittedly this if-else block looks more like java than swift code. If you could make this more ternay/swift-like, that'd be helpful. I've found that using the `if let` syntax doesn't really work in ForEach blocks